pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -959,3 +959,74 @@ def test_pass_datetimeindex_to_index(self):
         expected = Index(rng.to_pydatetime(), dtype=object)
 
         tm.assert_numpy_array_equal(idx.values, expected.values)
+
+
+@pytest.mark.parametrize(
+    "ts_input,fold",
+    [
+        (1572136200000000000, 0),
```

Comment:
Wouldn't it make more sense to use two parametrize decorators? One for `fold` and one for inputs?

I also think you should add at least one example of each type that could raise - int, float, np.datetime64, unambiguous string, naive datetime.

I do not think you should add a test for naive strings unless we are all agreed that "never accept fold with naive strings" is the right *semantics* (as opposed to, "We'd like to, but it's too hard").

Reply:
Done. Since we are writing this test, might as well make it complete. Not like it takes long to run.